src/azure-firewall/azext_firewall/custom.py

```diff
@@ -610,30 +610,31 @@ def update_azure_firewall_policies(cmd,
 
     # identity
     (ManagedServiceIdentityUserAssignedIdentitiesValue,
-     ManagedServiceIdentity) = cmd.get_models('ManagedServiceIdentityUserAssignedIdentitiesValue',
+     ManagedServiceIdentity) = cmd.get_models('Components1Jq1T4ISchemasManagedserviceidentityPropertiesUserassignedidentitiesAdditionalproperties',
                                               'ManagedServiceIdentity')
     if user_assigned_identity is not None:
         user_assigned_indentity_instance = ManagedServiceIdentityUserAssignedIdentitiesValue()
         user_assigned_identities_instance = dict()
         user_assigned_identities_instance[user_assigned_identity] = user_assigned_indentity_instance
         identity_instance = ManagedServiceIdentity(
             type="UserAssigned",
-            user_assigned_identities=user_assigned_identities_instance
+            Components1Jq1T4ISchemasManagedserviceidentityPropertiesUserassignedidentitiesAdditionalproperties=user_assigned_identities_instance
         )
         instance.identity = identity_instance
 
     return instance
 
 
 def set_azure_firewall_policies(cmd, resource_group_name, firewall_policy_name, parameters):
-    if parameters.identity is None:
-        ManagedServiceIdentity = cmd.get_models('ManagedServiceIdentity')
+    # Firewall Policy can't contain premium only properties - Identity
```

Comment:
Could you tell more context about this change?

Reply:
If the tier in this request is `Standard` and the identity  is not None, API will return Error response. Because it's just allowed when tier is `Premium`.